Versions v0.31.2-4 generates invalid code for cluster in bl602-pac

Using [this svd](https://github.com/sipeed/bl602-pac/blob/main/soc602_reg.svd) and svd2rust version 0.31.2, 0.31.3, or 0.31.4 generates a PAC that fails to build.
0.31.1 and 0.30.3 do generate a PAC that can successfully build.
This is the error returned:

```text
error[E0373]: closure may outlive the current function, but it borrows `self`, which is owned by the current function
   --> src/dma.rs:109:20
    |
109 |         (0..4).map(|n| unsafe {
    |                    ^^^ may outlive borrowed value `self`
110 |             &*(self as *const Self)
    |                ---- `self` is borrowed here
    |
note: closure is returned here
   --> src/dma.rs:109:9
    |
109 | /         (0..4).map(|n| unsafe {
110 | |             &*(self as *const Self)
111 | |                 .cast::<u8>()
112 | |                 .add(256)
113 | |                 .add(256 * n)
114 | |                 .cast()
115 | |         })
    | |__________^
help: to force the closure to take ownership of `self` (and any other referenced variables), use the `move` keyword
    |
109 |         (0..4).map(move |n| unsafe {
    |                    ++++
```

here is the generated code that it is referring to (or you can view it [here](https://github.com/9names/bl602-pac/blob/dfac4f8413d79b0707a02af4229374901db110ce/src/dma.rs#L108C26-L108C26)):
```rust
#[doc = "Iterator for array of:"]
    #[doc = "0x100..0x150 - Cluster CH%s, containing SrcAddr, DstAddr, LLI, Control, Config"]
    #[inline(always)]
    pub fn ch_iter(&self) -> impl Iterator<Item = &CH> {
        (0..4).map(|n| unsafe {
            &*(self as *const Self)
                .cast::<u8>()
                .add(256)
                .add(256 * n)
                .cast()
        })
    }
```

applying the fix in that error message is sufficient to get the PAC to build.

the cluster that is causing the issues is on [line 24279 of the svd](https://github.com/sipeed/bl602-pac/blob/64835f6349fc52e526b197be9e983c8718c29356/soc602_reg.svd#L24279), in case this is specific to the way the cluster is written.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,7 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
 
 ## [Unreleased]
 
-- Fix `cargo doc` constants generation 
+- `move` in `RegisterBlock::reg_iter` implementation (iterator of register/cluster array)
+- Fix `cargo doc` constants generation
 
 ## [v0.31.4] - 2024-01-03
 

--- a/src/generate/peripheral/accessor.rs
+++ b/src/generate/peripheral/accessor.rs
@@ -190,7 +190,7 @@ impl ToTokens for RawArrayAccessor {
             #[doc = #doc]
             #[inline(always)]
             pub fn #name_iter(&self) -> impl Iterator<Item=&#ty> {
-                (0..#dim).map(|n| #cast)
+                (0..#dim).map(move |n| #cast)
             }
         }
         .to_tokens(tokens);

